src/main/kotlin/no/nav/helse/flex/service/SykepengegrunnlagForNaeringsdrivende.kt

```diff
@@ -51,80 +54,106 @@ class SykepengegrunnlagForNaeringsdrivende(
+            val grunnbeloepSisteFemAar =
+                grunnbeloepService.hentHistorikk(soknad.startSykeforlop).takeIf { it.isNotEmpty() }
+                    ?: throw Exception("finner ikke historikk for g fra siste fem år")
+
+            val grunnbeloepPaaSykmeldingstidspunkt =
+                grunnbeloepSisteFemAar.find { it.dato.tilAar() == sykmeldingstidspunkt }!!.grunnbeløp
+
+            // TODO:: Rydd i variabelbruk etter testing.
+            val res =
+                hentPensjonsgivendeInntektForTreSisteArene(
+                    soknad.fnr,
+                    sykmeldingstidspunkt,
+                )
 
-        val justerteInntekter =
-            finnInntekterJustertFor6Gog12G(
-                inntekterJustertForGrunnbeloep,
-                grunnbeloepPaaSykmeldingstidspunkt,
+            val pensjonsgivendeInntekter = res?.filter { it.pensjonsgivendeInntekt.isNotEmpty() }
+
+            if (pensjonsgivendeInntekter.isNullOrEmpty()) {
+                return null
+            }
+
+            val grunnbeloepRelevanteAar =
+                finnGrunnbeloepForTreRelevanteAar(grunnbeloepSisteFemAar, pensjonsgivendeInntekter)
+
+            val inntekterJustertForGrunnbeloep =
+                finnInntekterJustertForGrunnbeloep(
+                    pensjonsgivendeInntekter,
+                    grunnbeloepRelevanteAar,
+                    grunnbeloepPaaSykmeldingstidspunkt,
+                )
+
+            val justerteInntekter =
+                finnInntekterJustertFor6Gog12G(
+                    grunnbeloepPaaSykmeldingstidspunkt,
+                    inntekterJustertForGrunnbeloep,
+                )
+            val gjennomsnittligInntektAlleAar = beregnGjennomsnittligInntekt(justerteInntekter)
+
+            return SykepengegrunnlagNaeringsdrivende(
+                gjennomsnittPerAar = justerteInntekter.map { AarVerdi(it.key, it.value.roundToBigInteger()) },
+                grunnbeloepPerAar = grunnbeloepRelevanteAar.map { AarVerdi(it.key, it.value) },
+                grunnbeloepPaaSykmeldingstidspunkt = grunnbeloepPaaSykmeldingstidspunkt,
+                beregnetSnittOgEndring25 = beregnEndring25Prosent(gjennomsnittligInntektAlleAar),
+                inntekter = pensjonsgivendeInntekter,
             )
-        val gjennomsnittligInntektAlleAar = beregnGjennomsnittligInntekt(justerteInntekter)
-
-        return SykepengegrunnlagNaeringsdrivende(
-            gjennomsnittPerAar = justerteInntekter.map { AarVerdi(it.key, it.value.roundToBigInteger()) },
-            grunnbeloepPerAar = grunnbeloepRelevanteAar.map { AarVerdi(it.key, it.value) },
-            grunnbeloepPaaSykmeldingstidspunkt = grunnbeloepPaaSykmeldingstidspunkt,
-            beregnetSnittOgEndring25 = beregnEndring25Prosent(gjennomsnittligInntektAlleAar),
-            inntekter = pensjonsgivendeInntekter,
-        )
+        } catch (e: Exception) {
+            log.error(e.message, e)
+            return null
```

Comment:
Stemmer det at ved å returnere null her vil vi ikke vise tall for bruker dersom Sigrun er nede? I så fall kunne vi latt den feile og heller havne i retry køen?

Reply:
Fixed, men lagt inn en en del TODOs for ytterligere rydding.